evalml/data_checks/outliers_data_check.py

```diff
@@ -56,11 +53,16 @@ def get_IQR(df, k=2.0):
             upper_bound = q3 + (k * iqr)
             return (lower_bound, upper_bound)
 
-        clf = IsolationForest(random_state=self.random_state)
-        clf.fit(X)
-        scores = pd.Series(clf.decision_function(X))
-        lower_bound, upper_bound = get_IQR(scores, k=2)
-        outliers = (scores < lower_bound) | (scores > upper_bound)
-        outliers_indices = outliers[outliers].index.values.tolist()
-        warning_msg = "Row '{}' is likely to have outlier data"
-        return [DataCheckWarning(warning_msg.format(row_index), self.name) for row_index in outliers_indices]
+        lower_bound, upper_bound = get_IQR(X)
+        indices = set()
+        # get the columns that fall out of the bounds, which means they contain outliers
+        for idx, bound in enumerate([lower_bound, upper_bound]):
+            cols_in_range = (X >= bound.values) if idx == 0 else (X <= bound.values)
+            cols_in_range = cols_in_range.all()
+            outlier_cols = cols_in_range[~cols_in_range].keys()
+            indices.update(outlier_cols.tolist())
+        # order the columns by how they appear in the dataframe
+        indices = sorted(list(indices), key=lambda x: X.columns.tolist().index(x))
+        warning_msg = "Column '{}' is likely to have outlier data"
+        s = [DataCheckWarning(warning_msg.format(row_index), self.name) for row_index in indices]
```

Comment:
@bchen1116 how does this code work? Why does this say `row_index` and not `column_name`?

Reply:
Why not save the column names directly in the impl? That would make the code simpler. Suggestion:
```python
def get_IQR(df, k=2.0):
    q1 = df.quantile(0.25)
    q3 = df.quantile(0.75)
    iqr = q3 - q1
    lower_bound = pd.Series(q1 - (k * iqr), name='lower_bound')
    upper_bound = pd.Series(q3 + (k * iqr), name='upper_bound')
    return pd.concat([lower_bound, upper_bound], axis=1)

iqr = get_IQR(X)
has_outliers = ((X < iqr['lower_bound']) & (X < iqr['upper_bound'])).any()
cols = has_outliers.index[has_outliers]
return [DataCheckWarning(warning_msg.format(col), self.name) for col in cols]
```
I love pandas bitwise operators!

Reply:
Ah, that was my mistake. I can fix, thanks for the simplification suggestion!